tools/binlog-slave-client.md

```diff
@@ -0,0 +1,149 @@
+  // for the `string` type: text longtext mediumtext char tinytext varchar
+  // blob longblog mediumblog binary tinyblob varbinary
+  // enum set
+  // for the `json` type: json
+  optional string mysql_type = 2 [ (gogoproto.nullable) = false ];
+  optional bool is_primary_key = 3 [ (gogoproto.nullable) = false ];
+}
+
+// `Row` stores the actual data of a row.
+message Row { repeated Column columns = 1; }
+
+// `MutationType` indicates the DML type.
+enum MutationType {
+  Insert = 0;
+  Update = 1;
+  Delete = 2;
+}
+
+// `Table` contains mutations in a table.
+message Table {
+  optional string schema_name = 1;
+  optional string table_name = 2;
+  repeated ColumnInfo column_info = 3;
+  repeated TableMutation mutations = 4;
+}
+
+// `TableMutation` stores mutations of a row.
+message TableMutation {
+  required MutationType type = 1;
+  // data after modification
+  required Row row = 2;
+  // data before modification. It only takes effect for `Update MutationType`.
+  optional Row change_row = 3;
+}
+
+// `DMLData` stores all the mutations caused by DML in a table.
+message DMLData {
+  // `tables` contains all the table changes.
+  repeated Table tables = 1;
+}
+
+// `DDLData` stores the DDL information.
+message DDLData {
+  // the database used currently
+  optional string schema_name = 1;
+  // the relates table
+  optional string table_name = 2;
+  // `ddl_query` is the original DDL statement query.
+  optional bytes ddl_query = 3;
+}
+
+// `BinlogType` indicates the binlog type, including DML and DDL.
+enum BinlogType {
+  DML = 0; //  Has `dml_data`
+  DDL = 1; //  Has `ddl_query`
+}
+
+// `Binlog` stores all the changes in a transaction. Kafka stores the serialized result of the structure data.
+message Binlog {
+  optional BinlogType type = 1 [ (gogoproto.nullable) = false ];
+  optional int64 commit_ts = 2 [ (gogoproto.nullable) = false ];
+  // `dml_data` is marshalled from the DML type.
+  optional DMLData dml_data = 3;
+  optional DDLData ddl_data = 4;
+}
+```
+
+For the definition of the data format, see [`binlog.proto`](https://github.com/pingcap/tidb-tools/blob/master/tidb_binlog/slave_binlog_proto/proto/binlog.proto).
+
+### Driver
+
+TiDB-Tools provides Driver which is used to read the binlog data in Kafka. It has the following features:
+
+* Read the Kafka data.
+* Locate the binlog stored in Kafka based on `commit ts`.
+
+You need to configure the following information when using Driver:
+
+* `KafkaAddr`: the address of the Kafka cluster
+* `CommitTS`: from which `commit ts` to start reading the binlog
+* `Offset`: from which Kafka `offset` to start reading data. If `CommitTS` is set, you needn't configure this parameter
+* `ClusterID`: the cluster ID of the TiDB cluster
+
+You can use Driver by quoting the Driver code in package and refer to the example code provided by Driver to learn how to use Driver and parse the binlog data. 
+
+Currently, there are two examples:
```

Comment:
Generally, limit the use of the "there be" pattern.

-> Currently, two examples are provided: